packages/migrate/src/__tests__/DbPull.test.ts

```diff
@@ -1170,15 +1170,191 @@ describe('postgresql views introspection warnings', () => {
+      ),
+    }
+    return setupParams
+  }
+
+  const setupPostgressForWarning = (warningCode: number, variant?: number) => {
+    const setupParams = computeSetupParams(warningCode, variant)
+
+    beforeEach(async () => {
+      await setupPostgres(setupParams)
+
+      // Back to original env vars
+      process.env = { ...originalEnv }
+      // Update env var because it's the one that is used in the schemas tested
+      process.env.TEST_POSTGRES_URI_MIGRATE = connectionString
+    })
+
+    afterEach(async () => {
+      // Back to original env vars
+      process.env = { ...originalEnv }
+      await tearDownPostgres(setupParams).catch((e) => {
+        console.error(e)
+      })
+    })
+  }
+
+  describe('postgresql inhereited 28/1 - inherited table found', () => {
+    const warningCode = 28
+    const variant = 1
+    setupPostgressForWarning(warningCode, variant)
+
+    test('basic introspection', async () => {
+      ctx.fixture(`introspection/postgresql/inherited-warning-${warningCode}-${variant}`)
+      const introspect = new DbPull()
+      const result = introspect.parse(['--print'])
+      await expect(result).resolves.toMatchInlineSnapshot(``)
+      expect(ctx.mocked['console.log'].mock.calls.join('\n')).toMatchInlineSnapshot(`
+        generator client {
+          provider = "prisma-client-js"
+        }
+
+        datasource db {
+          provider = "postgres"
+          url      = env("TEST_POSTGRES_URI_MIGRATE")
+        }
+
+        model capitals {
+          name       String?
+          population Float?  @db.Real
+          elevation  Int?
+          state      String  @unique @db.Char(2)
+        }
+
+        /// This table has subclasses and requires additional setup for migrations. Visit https://pris.ly/d/table-inheritance for more info.
+        /// The underlying table does not contain a valid unique identifier and can therefore currently not be handled by the Prisma Client.
+        model cities {
+          name       String?
+          population Float?  @db.Real
+          elevation  Int?
+
+          @@ignore
+        }
+
+
+        // introspectionSchemaVersion: NonPrisma,
+      `)
+      expect(ctx.mocked['console.info'].mock.calls.join('\n')).toMatchInlineSnapshot(``)
+      expect(ctx.mocked['console.error'].mock.calls.join('\n')).toMatchInlineSnapshot(`
+
+        // *** WARNING ***
+        // 
+        // The following models were ignored as they do not have a valid unique identifier or id. This is currently not supported by the Prisma Client.
+        // - "cities"
+        // 
+        // These tables are inherited tables, which are not yet fully supported.
+        // - Model "cities"
+        // 
+      `)
+      expect(ctx.mocked['process.stdout.write'].mock.calls.join('\n')).toMatchInlineSnapshot(``)
+      expect(ctx.mocked['process.stderr.write'].mock.calls.join('\n')).toMatchInlineSnapshot(``)
+    })
+  })
+
+  describe('postgresql inhereited 28/2 - inherited tables found', () => {
```

Comment:
What is this really testing? From a quick look, it seems more like the same but with the tables duplicated?

Reply:
It's just effectively making sure that stuff is being formatted correctly for when we detect multiple -- off the back of what happened previously 😅 but it works so am also happy to just not leave this in

Reply:
No problem - just the name might explain it a bit better.